### ISSUE TITLE
SPICE Ingester giving files incorrect permissions

### ISSUE THREAD

**Issue by bryan-harter** [MAVENSDC/maven-sdc#62]
As far as I can tell, this is specific to the SPICE ingester. The other ANC ingesters and the dropbox ingesters work fine.

But the SPICE ingester gives 640 permissions, when we need 644.  The code looks accurate, so we probably need to play around with it to figure out what is going on.

**Comment by torimarbois** [comment]
This is within the sdc server. The maven-sdc/scripts/ingest_anc_files/poc_ingest.sh holds the scripts for the cron tab that moves the spice files and theoretically adds the right permissions for everyone to view it.
To investigate, consider using print statements or a debug mode.

**Comment by bryan-harter** [comment]
So here is what I had been doing to debug:

1. We go into /maven/mavenpro/maven-sdc and `git checkout `fixing_spice_permissions'
2. On the branch `fixing_spice_permissions`, I currently have a few print statements that will print out into the console
3. Move the file mvn_app_red_241217_v01.bc back into the dropbox. This command should work: `mv /maven/data/anc/spice/ck/mvn_app_red_241217_v01.bc /maven/poc/spice/mvn_app_red_241217_v01.bc`. It doesn't need to be this file, this just happens to be the latest file at the time I'm writing this, and it still has bad permissions.
4. Run the command for poc-ingest.sh as seen in our crontab. NOTE: You'll need to export all of the variables at the top of the crontab before the command works. In otherwords, `export VENV_ACTIVATE=/maven/mavenpro/venvs/maven_prod/bin/activate` and `export SCRIPTS_DIRECTORY=/maven/mavenpro/maven-sdc/scripts`, etc. And then run `export MAVEN_PYTHON_ENV=production_update;source $VENV_ACTIVATE && $SCRIPTS_DIRECTORY/ingest_anc_files/poc_ingest.sh`. 


As you make changes to the fixing_spice_permissions branch, i.e. adding more print statements or minor tweaks, just call git fetch to pull down the latest changes. 

And then when you're done testing, just be sure to `git checkout main` again, and no one will be the wiser.

**Comment by bryan-harter** [comment]
Here are some random ideas I had:

- Make sure the permissions are actually what we expect. If not, determine where they get set and why they aren't set to 755
- Try setting the permissions to anything else (like 777) and see if that actually changes anything 
- Maybe there is something wrong with the linux user or the crontab permissions that don't enable setting those file permissions? I don't even know if that's a thing. 
- What if we put the file in the dropbox with 755 permissions already? Does it still get moved into /spice/ck with 640 permissions? If not, can we chmod before we move the files and that'll solve the issue?

**Comment by torimarbois** [comment]
Sean forwarded us an email from Paul W. from Boston University, who is having file access issues. I wonder if that's related to this open issue?

> Hello again,
> 
> Sorry to be giving you a running commentary on my day, but here's my
> latest glitch.
> 
> https://lasp.colorado.edu/maven/sdc/team/data/anc/spice/spk/?C=M;O=A
> 
> I'm looking to download the medium-range kernels for planning purposes.
> 
> trj_orb_22433-22435_23337_v1.bsp        2024-11-18 18:45         24M    
> This downloads and works just fine.
> 
> 
> trj_orb_22618-22619_23518_v1.bsp        2024-12-16 18:46         24M    
> "Forbidden
> 
> You don't have permission to access this resource."
> 
> ...
> 
> I tested out a few other files in this folder. I get similar errors for
> all files created after about 9 December or so, earlier files are fine.
> 
> Happy New Year!
> Paul
> 

> > On 1/3/25 11:22, Paul Withers wrote:
> > Never mind, it's back to normal for me now.
> >
> > Cheers,
> > Paul
> >

>> > On 1/3/25 10:40, Paul Withers wrote:
> >> Hi Sean,
> >>
> >> I get the following error message when I try to access the SDC.
> >>
> >> Proxy Error
> >>
> >> The proxy server received an invalid response from an upstream server.
> >> The proxy server could not handle the request
> >>
> >> Reason: Error reading from remote server
> >>
> >>
> >> ...
> >>
> >>
> >> The public site also has problems for me.
> >>
> >> HTTP/1.1 429 Too Many Requests
> >> Content-Type: text/html
> >> Retry-After: 300
> >>
> >> ...
> >>
> >> Cheers,
> >> Paul

**Comment by bryan-harter** [comment]
Yup it must be! Whatever is causing the issues must apply to all spice files. So it must be an issue somewhere in the spice ingester that is not present in the anc or l0 ingester
